### PR TITLE
Fix conflict with Simplex::GridGenerator

### LINKED ISSUE #10765: doxygen is confused by simplex namespace
subdivided_hyper_rectangle now also contains the documentation for Simplex::GridGenerator::subdivided_hyper_rectangle at the bottom:
https://www.dealii.org/developer/doxygen/deal.II/namespaceGridGenerator.html#ac76417d7404b75cf53c732f456e6e971
(grep for "tetrahedral")
I am not sure why that happens. Maybe we can change use a different namespace for simplex stuff?

Also, shouldn't simplex/grid_generator.h also be guarded by the simplex define?

### PR DESCRIPTION
**Pull request by peterrum**
closes #10765.

### CODE REVIEW

File: include/deal.II/simplex/grid_generator.h [hunk 1]
```diff
@@ -27,221 +27,217 @@
 
 DEAL_II_NAMESPACE_OPEN
 
-namespace Simplex
+namespace GridGenerator
```
Comment (tjhei) [inline include/deal.II/simplex/grid_generator.h:30]: should there be a ``#ifdef DEAL_II_WITH_SIMPLEX_SUPPORT`` here?

**Comment by tjhei** [comment]
I prefer this over the old namespacing. :+1:

File: include/deal.II/simplex/grid_generator.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/simplex/grid_generator.h:30]: Do you like this solution?

**Comment by tjhei** [comment]
I am not sure I understand. I thought that we don't want simplex related functions to show up in doxygen right now. As a consequence, you will need to ``#ifdef`` based on simplex support. Or am I missing something?

**Comment by peterrum** [comment]
I don't really mind if it appears in doxygen as long as there are enough asserts which prevent the usage.

**Comment by bangerth** [comment]
But things that we don't intend to enable yet shouldn't show up in the documentation either. That would just be confusing.

File: include/deal.II/simplex/grid_generator.h [hunk 2]
```diff
@@ -27,221 +27,225 @@
...
-   * This namespace provides a collection of functions to generate simplex
-   * triangulations for some basic geometries.
+   * Create a coordinate-parallel brick from the two diagonally opposite
+   * corner points @p p1 and @p p2. The number of vertices in coordinate
+   * direction @p i is given by <tt>repetitions[i]+1</tt>.
    *
-   *  @ingroup simplex
+   * @note This function connects internally 4/8 vertices to quadrilateral/
+   *   hexahedral cells and subdivides these into 2/5 triangular/
+   *   tetrahedral cells.
```
Comment (bangerth) [inline include/deal.II/simplex/grid_generator.h:39]: This is going to create no space before but one space after the `/` when rendered. Just put the first word of lines 38 and 39 at the end of the previous line and then let indentation break where necessary.

File: include/deal.II/simplex/grid_generator.h [hunk 3]
```diff
@@ -27,221 +27,225 @@
...
-  namespace GridGenerator
+  template <int dim, int spacedim>
+  void
+  subdivided_hyper_rectangle_with_simplices(
+    Triangulation<dim, spacedim> &   tria,
+    const std::vector<unsigned int> &repetitions,
+    const Point<dim> &               p1,
+    const Point<dim> &               p2,
+    const bool                       colorize = false)
   {
```
Comment (bangerth) [inline include/deal.II/simplex/grid_generator.h:53]: What's the reason to have the implementation of this function in the .h file, by the way?

**Comment by peterrum** [comment]
I have moved the implementation in a separate file, added the ifdefs, and fixed the doxygen comment. Thanks for the comments!

**Comment by drwells** [comment]
The new name is verbose but clear and avoids doxygen problems.

**Comment by drwells** [comment]
@tjhei @bangerth are you two happy with this change? If so lets merge!

**Pull request merged**